feature_engine/transformation/log.py

```
@@ -164,3 +164,206 @@ def _more_tags(self):
+        """
+        Transforms the variables using log transformation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame
+        ValueError
+            - If the variable(s) contain null values
+            - If the df has different number of features than the df used in fit()
+            - If some variables contains zero or negative values after suming constant C
+
+        Returns
+        -------
+        X: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = super().transform(X)
+
+        # check contains zero or negative values
+        if (X[self.variables_] + self.C_ <= 0).any().any():
+            raise ValueError(
+                "Some variables contain zero or negative values after suming"
+                + "constant C, can't apply log"
+            )
+
+        # transform
+        if self.base == "e":
+            X.loc[:, self.variables_] = np.log(X.loc[:, self.variables_] + self.C_)
+        elif self.base == "10":
+            X.loc[:, self.variables_] = np.log10(X.loc[:, self.variables_] + self.C_)
+
+        return X
+
+    def inverse_transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the variables using exponential transformation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be detransformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame
+        ValueError
+            - If the variable(s) contain null values
+            - If X has different number of features than X used in fit()
+            - If some variables contains zero or negative values
```

I don't think we should raise this error, because log of a value >0 and <1 will be negative, and can return the inverse. Am I correct?

---

got it, additionally as the variable is already in log scale, the exponential function accepts negative and positive values, so there's no need for this error, thanks for catching it.